common/config/azure-pipelines/templates/publish.yaml

```diff
@@ -4,9 +4,15 @@ parameters:
   - name: BranchName
     type: string
     default: $(Build.SourceBranchName)
+  - name: PublishVersionTag
+    type: string
+    default: latest
 
 steps:
-  - script: 'node common/scripts/install-run-rush.js publish --apply --publish --include-all --target-branch ${{ parameters.BranchName }} --add-commit-details --set-access-level public'
+  - script: 'node common/scripts/install-run-rush.js publish --apply --pack --release-folder $(Build.ArtifactStagingDirectory)/packages --include-all --target-branch ${{ parameters.BranchName }} --add-commit-details'

```

Comment:
looks like you need to keep `--publish` else it will do a dry run.
`--publish` with `--pack` will create archives instead of publishing to registry.
Logic: https://github.com/microsoft/rushstack/blob/ffa49929e54221d1e5f766a23008c12fd7ec7345/libraries/rush-lib/src/cli/actions/PublishAction.ts#L398-L410